### ISSUE TITLE
HTTP error 404 for PyYAML-4.1-cp35-cp35m-linux_armv7l.whl

### ISSUE THREAD

**Issue by samuelharmer** [bennuttall/piwheels#73]
The package is listed [here](https://www.piwheels.org/simple/pyyaml/), but it gives a 404.

```
Collecting PyYAML>=3.0 (from flask-swagger==0.2.13->apache-airflow[crypto,postgres])
  HTTP error 404 while getting https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv7l.whl#sha256=a5b125a24244830d574ccfd8fab861198d2ffe491b73179ffd75abf78caee6ff (from https://www.piwheels.org/simple/pyyaml/)
  Could not install requirement PyYAML>=3.0 from https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv7l.whl#sha256=a5b125a24244830d574ccfd8fab861198d2ffe491b73179ffd75abf78caee6ff (from flask-swagger==0.2.13->apache-airflow[crypto,postgres]) because of error 404 Client Error: Not Found for url: https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv7l.whl
Could not install requirement PyYAML>=3.0 from https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv7l.whl#sha256=a5b125a24244830d574ccfd8fab861198d2ffe491b73179ffd75abf78caee6ff (from flask-swagger==0.2.13->apache-airflow[crypto,postgres]) because of HTTP error 404 Client Error: Not Found for url: https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv7l.whl for URL https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv7l.whl#sha256=a5b125a24244830d574ccfd8fab861198d2ffe491b73179ffd75abf78caee6ff (from https://www.piwheels.org/simple/pyyaml/)
```

**Comment by samuelharmer** [comment]
As a workaround, the following satisfied `pip install apache-airflow` but it would be great if this package would install via `pip`.

Build and install [LibYAML][libyaml]
```
$ mkdir ~/libyaml && cd ~/libyaml
$ wget http://pyyaml.org/download/libyaml/yaml-0.2.1.tar.gz
$ tar -zxvf yaml-0.2.1.tar.gz && cd yaml-0.2.1/
$ ./configure
$ make
# make install
```

Build and install [PyYAML][pyyaml]
```
$ mkdir ~/pyyaml && cd ~/pyyaml
$ wget http://pyyaml.org/download/pyyaml/PyYAML-3.12.tar.gz
$ tar -zxvf PyYAML-3.12.tar.gz && cd PyYAML-3.12
(venv) $ python setup.py --with-libyaml install
```

[libyaml]: https://pyyaml.org/wiki/LibYAML
[pyyaml]: https://pyyaml.org/

**Comment by samuelharmer** [comment]
```
$ cat /proc/device-tree/model
Raspberry Pi 3 Model B Plus Rev 1.3

$ cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 9 (stretch)"
NAME="Raspbian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=raspbian
ID_LIKE=debian
HOME_URL="http://www.raspbian.org/"
SUPPORT_URL="http://www.raspbian.org/RaspbianForums"
BUG_REPORT_URL="http://www.raspbian.org/RaspbianBugs"

$ gcc --version
gcc (Raspbian 6.3.0-18+rpi1+deb9u1) 6.3.0 20170516
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

**Comment by tsangiotis** [comment]
I can confirm this for all 4.1 distributions. 
Another solution is to use the previous stable versions (3.12 in my case).

**Comment by uSpike** [comment]
This is affecting me as well.  Is there a reason this distribution was removed?  It is interesting that it still shows up on the "simple" page: https://www.piwheels.org/simple/pyyaml/

    PyYAML-4.1-cp35-cp35m-linux_armv6l.whl
    PyYAML-4.1-cp35-cp35m-linux_armv7l.whl
    PyYAML-4.1-cp34-cp34m-linux_armv7l.whl
    PyYAML-4.1-cp34-cp34m-linux_armv6l.whl

**Comment by uSpike** [comment]
So it seems that PyYAML devs retracted that release.  It seems that piwheels detected that and removed the files, but *not* the directories that show up on the `/simple/` path.  This breaks tools like `pipenv`:

```
Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  You can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
404 Client Error: Not Found for url: https://www.piwheels.org/simple/pyyaml/PyYAML-4.1-cp35-cp35m-linux_armv6l.whl
```

which will try to follow those links, even if you specify a different version in your `Pipfile`.

Piwheels devs, could you *please* consider deleting those directories?

**Comment by ahunt-xx** [comment]
I have been trying to install python-escpos and it keeps trying to install the PyYAML-4.1 and getting the 404. I did an install of a different version of PyYAML but it keeps trying to install the 4.1. 

Any help would be greatly appreciated.

**Comment by NyanKiyoshi** [comment]
It looks like deleted releases from PyPI are not correctly handled by piwheels (https://github.com/yaml/pyyaml/issues/203).

**Comment by bennuttall** [comment]
Sorry for the lack of response here. For some reason github isn't sending me notifications for issues in this repo.

I did see it a few days ago and I purged the 4.1 releases, but the index hadn't been updated so they were still showing in the list (and 404ing).

Now I have forced the index to be rebuilt, and it looks ok now. Can anyone confirm it's all working ok?

The current procedure for deletions is they must be reported to us if they need removing, we won't pick them up automatically until we introduce logic for this in the system.
